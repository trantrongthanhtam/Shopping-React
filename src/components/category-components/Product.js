import React from "react";
import { Link } from "react-router-dom";
import styles from "../../css/category.module.css";

class Product extends React.Component {
	render() {
		const { src, id, price, onAddToCart } = this.props;
		return (
			<div className={styles.product}>
				<Link to="/detail">
					<div className={styles.hide}>
						<button
							onClick={onAddToCart.bind(id)}
							className={styles.addCart}
						>
							ADD TO CART
						</button>
						<img src={src} alt="product" />
					</div>
					<h3>{id}</h3>
					<p>{price}</p>
				</Link>
			</div>
		);
	}
}

export default Product;
