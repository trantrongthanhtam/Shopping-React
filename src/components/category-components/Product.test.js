import React from "react";
import renderer from "react-test-renderer";
import Product from "./Product.js";
const mockProduct = {
	/*...*/
};
test("Product component", () => {
	const component = renderer.create(
		// <Product {...mockProduct} />
		<div>Hello</div>
	);
	const tree = component.toJSON();
	expect(tree).toMatchSnapshot();
});
