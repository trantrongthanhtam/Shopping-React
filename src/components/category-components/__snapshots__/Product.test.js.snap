// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Product component 1`] = `
<div>
  Hello
</div>
`;
